airbyte-integrations/connectors/source-gcs/unit_tests/test_cursor.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+from airbyte_cdk.sources.file_based.stream.cursor import DefaultFileBasedCursor
+
+
+def test_add_file_successfully(cursor, remote_file, logger):
+    cursor.add_file(remote_file)
+
+    assert len(cursor._file_to_datetime_history) == 1
+
+
+def test_add_file_successfully_with_full_history(cursor, remote_file, remote_file_b, logger):
+    DefaultFileBasedCursor.DEFAULT_MAX_HISTORY_SIZE = 1
+
+    cursor.add_file(remote_file)
+    cursor.add_file(remote_file_b)
+
+    assert len(cursor._file_to_datetime_history) == 1
+
+
+def test_add_file_with_no_history(cursor, remote_file, remote_file_b, logger):
+    DefaultFileBasedCursor.DEFAULT_MAX_HISTORY_SIZE = 1
+    cursor.add_file(remote_file)
+    cursor.add_file(remote_file_b)
+
+    assert len(cursor._file_to_datetime_history) == 1
```

Comment:
I'm confused, what is the difference between these two tests?

Reply:
Good catch, I guess I forgot to set history to 0